src/infra/blockchain_client/blockchain_client.rs

```diff
@@ -125,7 +125,7 @@ impl BlockchainClient {
     pub async fn fetch_listening(&self) -> anyhow::Result<()> {
         tracing::debug!("fetching listening status");
 
-        let result = self.http.request::<bool, Vec<()>>("net_listening", vec![]).await;
+        let result = self.http.request::<bool, _>("net_listening", [(); 0]).await;
```

Comment:
FWIW empty Vec's aren't allocated in the heap.

Reply:
Yeah I thought of that when writting, but didn't make it explicit. My intention was to  use the same pattern on every call for consistency and maybe to hint something like copilot not to use vecs in the future.

Reply:
In fact, `Vec<()>` never allocates as () is zero-sized